.github/workflows/build-package.yml

```diff
@@ -73,10 +74,31 @@ jobs:
           sparse-checkout: ${{ inputs.package_name }}
           path: ${{ github.workspace }}/packages
 
+      - name: Enable Ccache
+        id: enable-ccache
+        run: |
+          `cygpath "${{ github.workspace }}"`/.github/scripts/enable-ccache.sh
+
+      - name: Restore Ccache
+        uses:  actions/cache/restore@v4
+        with:
+          path: ${{ github.workspace }}/ccache
+          key: main-ccache-${{ steps.enable-ccache.outputs.timestamp }}
+          restore-keys: main-ccache-
+
       - name: Build ${{ inputs.package_name }}
         working-directory: ${{ github.workspace }}/packages/${{ inputs.package_name }}
         run: |
+          ccache -svv  || true
           `cygpath "${{ github.workspace }}"`/.github/scripts/build-package.sh ${{ inputs.packages_repository }}
+          ccache -svv || true
+
+      - name: Save Ccache
+        if: always()
```

Comment:
why is this condition needed?

Reply:
One want's to save ccache even when build fails.